### PR TITLE
KODI: Update NEON support

### PR DESCRIPTION
**Pull request by C-Prime90**
## Description
This commit makes CCPUInfo::HasNeon() report NEON enabled on Aarch64 targets and updates the ifdef's for newer preprocessor.

## How Has This Been Tested?
This has bin tested locally on Amlogic S812/S905 running on EmbER firmware.

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by stefansaraev** [comment]
are you 100% that enabling -funsafe-math-optimizations is safe for kodi ?

**Comment by croniccorey** [comment]
I've had it enabled in my Linux builds since 16.0 without issues, i can remove that if you'd prefer.

File: cmake/scripts/linux/ArchSetup.cmake [hunk 1]
```diff
@@ -16,10 +16,11 @@ else()
   elseif(CPU MATCHES arm)
     set(ARCH arm)
     set(NEON True)
-    set(NEON_FLAGS "-mfpu=neon -mvectorize-with-neon-quad")
+    set(NEON_FLAGS "-mfpu=neon -funsafe-math-optimizations -mvectorize-with-neon-quad")
   elseif(CPU MATCHES aarch64 OR CPU MATCHES arm64)
     set(ARCH aarch64)
-    set(NEON False)
+    set(NEON True)
+    set(NEON_FLAGS "-march=armv8-a -funsafe-math-optimizations")
```
Comment (stefansaraev) [inline cmake/scripts/linux/ArchSetup.cmake]: aarch64 is always neon. you dont need these, and you should not override user's specified -march= anyway

**Comment by croniccorey** [comment]
I've updated the PR based on your comments.

**Comment by MartijnKaijser** [comment]
@FernetMenta @peak3d ping as well

**Comment by FernetMenta** [comment]
What non-NEON systems do we support? I am wondering if we can just drop the ifdef and make it default.

**Comment by stefansaraev** [comment]
@FernetMenta rpi1/0

**Comment by FernetMenta** [comment]
so we enable neon with a compiler switch, and then test for preprocessor macros? seems a bit combersome and error prone to me. finally results in this kind of issues.
why don't we user something like HAS_NEON that is set by defult and can be disabled for the outdated hardware

**Comment by wsnipex** [comment]
agreed, we should be using the preprocessor defines here.

**Comment by MartijnKaijser** [comment]
@croniccorey could you do the requested changes?

**Comment by croniccorey** [comment]
so you would like to drop the compiler switch in favor of preprocessor defines?

**Comment by wsnipex** [comment]
>You would like to drop the compiler switch in favor of preprocessor defines or drop preprocessor defines in favor of HAS_NEON set by cmake?
yes, this please

**Comment by croniccorey** [comment]
Ok i will update PR to use HAS_NEON

**Comment by croniccorey** [comment]
@MartijnKaijser @FernetMenta @wsnipex I have updated the PR based on your comments

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
@croniccorey thx for sticking with us and for your contribution

**Pull request merged**